diff --git a/node_modules/react-native-phone-number-input/lib/index.d.ts b/node_modules/react-native-phone-number-input/lib/index.d.ts
index cd625e5..c792df3 100644
--- a/node_modules/react-native-phone-number-input/lib/index.d.ts
+++ b/node_modules/react-native-phone-number-input/lib/index.d.ts
@@ -14,6 +14,7 @@ export interface PhoneInputProps {
   value?: string;
   defaultValue?: string;
   disabled?: boolean;
+  disableCountryCode?:boolean;
   disableArrowIcon?: boolean;
   placeholder?: string;
   onChangeCountry?: (country: Country) => void;
@@ -37,6 +38,7 @@ export interface PhoneInputState {
   modalVisible: boolean;
   countryCode: CountryCode;
   disabled: boolean;
+  disableCountryCode:boolean
 }
 export default class PhoneInput extends Component<
   PhoneInputProps,
diff --git a/node_modules/react-native-phone-number-input/lib/index.js b/node_modules/react-native-phone-number-input/lib/index.js
index b89c388..80b8aeb 100644
--- a/node_modules/react-native-phone-number-input/lib/index.js
+++ b/node_modules/react-native-phone-number-input/lib/index.js
@@ -27,6 +27,7 @@ export default class PhoneInput extends PureComponent {
       modalVisible: false,
       countryCode: props.defaultCode ? props.defaultCode : "IN",
       disabled: props.disabled || false,
+      disableCountryCode:props.disableCountryCode || false
     };
   }
 
@@ -48,6 +49,16 @@ export default class PhoneInput extends PureComponent {
     }
   }
 
+  async componentDidUpdate(prevProps){
+      if(prevProps.defaultCode !=this.props.defaultCode){
+            const {defaultCode} = this.props
+            if(defaultCode){
+          const code = await getCallingCode(defaultCode);
+                    this.setState({ code ,countryCode:defaultCode});
+            }
+            }
+  }
+
   getCountryCode = () => {
     return this.state.countryCode;
   };
@@ -109,12 +120,12 @@ export default class PhoneInput extends PureComponent {
 
   getNumberAfterPossiblyEliminatingZero() {
     let { number, code } = this.state;
-    if (number.length > 0 && number.startsWith("0")) {
-      number = number.substr(1);
+    // if (number.length > 0 && number.startsWith("0")) {
+    //   number = number.substr(1);
+    //   return { number, formattedNumber: code ? `+${code}${number}` : number };
+    // } else {
       return { number, formattedNumber: code ? `+${code}${number}` : number };
-    } else {
-      return { number, formattedNumber: code ? `+${code}${number}` : number };
-    }
+    // }
   }
 
   renderDropdownImage = () => {
@@ -160,7 +171,7 @@ export default class PhoneInput extends PureComponent {
       countryPickerButtonStyle,
       layout = "first",
     } = this.props;
-    const { modalVisible, code, countryCode, number, disabled } = this.state;
+    const { modalVisible, code, countryCode, number, disabled,disableCountryCode } = this.state;
     return (
       <CountryModalProvider>
         <View
@@ -177,7 +188,7 @@ export default class PhoneInput extends PureComponent {
               flagButtonStyle ? flagButtonStyle : {},
               countryPickerButtonStyle ? countryPickerButtonStyle : {},
             ]}
-            disabled={disabled}
+            disabled={disableCountryCode}
             onPress={() => this.setState({ modalVisible: true })}
           >
             <CountryPicker
@@ -188,7 +199,7 @@ export default class PhoneInput extends PureComponent {
               filterProps={filterProps}
               countryCode={countryCode}
               withCallingCode
-              disableNativeModal={disabled}
+              disableNativeModal={disableCountryCode}
               visible={modalVisible}
               theme={withDarkTheme ? DARK_THEME : DEFAULT_THEME}
               renderFlagButton={this.renderFlagButton}
